client/src/components/authentication/Signup.js

```diff
@@ -1,32 +1,49 @@
-// import React, { useRef } from "react";
-// import { Form, Button, Card } from "react-bootstrap";
+import React, { useCallback, useRef } from "react";
+import { Form, Button, Card } from "react-bootstrap";
+import app from "./Base";
 
-// export default function Signup() {
-//     const emailRef = useRef();
-//     const passwordRef = useRef();
-//     const passwordConfirmRef = useRef();
 
-//     return (
-//         <>
-//         <Card>
-//             <Card.Body>
-//                 <h3 className="text-center mb-4">Sign Up</h3>
-//                 <Form>
-//                     <Form.Group id="email">
-//                         <Form.Label>Email</Form.Label>
-//                         <Form.Control type="email" ref={emailRef} required />
-//                         <Form.Group id="email">
-//                         <Form.Label>Email</Form.Label>
-//                         <Form.Control type="email" ref={emailRef} required />
-//                         <Form.Group id="password">
-//                         <Form.Label>Password</Form.Label>
-//                         <Form.Control type= "password" ref={passwwordRef} required />
-//                     </Form.Group>
-//                     <Form.Group id="email">
-//                         <Form.Label>Confirm Password</Form.Label>
-//                         <Form.Control type="password-confirm" ref={confirmPasswordRef} required />
-//                 </Form>
-//             </Card.Body>
-//         </Card>
-//     )
-// }
\ No newline at end of file
+const Signup = ({ history }) => {
+    const handleSignUp = useCallback{async event => {
+        event.preventDefault();
+        const { email, password } = event.target.elements;
+        try {
+            await app
+            .auth()
+            .createUserWithEmailAndPassword(email.value, password.value);
+            history.pushState("/");
+        } catch (error) {
+            alert(error);
+                }
+    }, [history])}
+
```

Comment:
Like the login functions, these are two separate signup functions. You should be combining them into one signup function. useRef values will be retrieved with emailRef.current.value and passwordRef.current.value.